Use native file dialogs in Windows

lwchkg opened the pull request:
Switch to native file dialogs in Windows. The following dialogs are switched:
- Game open dialog
- Game save dialog
- Screenshot save dialog
- ASCII art save dialog

Suggesting filenames during file saves are not implemented because QtQuick.Dialogs.FileDialog does not support this.

Also, since the FileDialog QML in Qt 6.3 is incompatible with those in Qt 5.15, only Qt 6.2 or up is supported. (I have no information whether it works for Qt 6.0 or 6.1.)

enz committed:
Use native file open dialog in Windows

enz commented on pentobi/qml/Main.qml:
```diff
@@ -170,7 +170,7 @@ ApplicationWindow {
     DialogLoader { id: initialRatingDialog; url: "InitialRatingDialog.qml" }
     DialogLoader { id: newFolderDialog; url: "NewFolderDialog.qml" }
-    DialogLoader { id: openDialog; url: "OpenDialog.qml" }
+    DialogLoader { id: openDialog; url: (Logic.useNativeDialog() ? "OpenDialogNative.qml" : "OpenDialog.qml") }
```
Sure, you decide. Although I would suggest a property bool useNativeFileDialog. Javascript functions should be avoided in QML unless really needed.

lwchkg committed:
Use native save & export dialogs in Windows build

lwchkg commented:
Now the patch is ready. You may want to resolve that 96-column line issue before merging. BTW, you may also want to test if this patch breaks any of the 4 file dialogs for Linux.

As in the updated description, suggesting filenames for save operation is not supported in QtQuick.Dialogs.FileDialog in Qt 6.3 (was supported in Qt 6.2, but the new version removed the feature). The function appears to be available in the Qt Labs Platform dialog, but I am not comfortable to use it because Labs is experimental.

Please also advice if you need me to sign a CLA (contributor license agreement).

enz commented:
I tested this on Ubuntu 22.04 (Qt 5.15.3) and nothing seems to be broken.

Could you replace my name with yours in the file headers of all newly created files? I don't think that the changes to existing files are substantial enought yet to constitute a changed copyright there, but if you submit major changes to existing files in the future, just add a second @author line with your name.

CLA is not needed.

lwchkg committed:
Update author information on new files

lwchkg commented:
Thanks for your comment. It's a good news that nothing is broken.

I have updated the author names in the new files, and added my name in `AUTHORS.md` as an contributor.

enz commented:
Squash-merged into master. Thanks for your work.
